ext/strings_test.go

```diff
@@ -329,3 +330,229 @@ func TestStringsWithExtension(t *testing.T) {
 		t.Fatalf("env.Extend(Strings()) failed: %v", err)
 	}
 }
+
+func TestStringFormat(t *testing.T) {
+	tests := []struct {
+		name   string
+		format string
+		//formatArgs     []ref.Val
+		formatArgs     string
+		locale         string
+		err            string
+		expectedOutput string
+	}{
+		{
+			name:           "no-op",
+			format:         "no substitution",
+			expectedOutput: "no substitution",
+		},
+
+		{
+			name:           "mid-string substitution",
+			format:         "str is %s and some more",
+			formatArgs:     `"filler"`,
+			expectedOutput: "str is filler and some more",
+		},
+		{
+			name:           "percent escaping",
+			format:         "%% and also %%",
+			expectedOutput: "% and also %",
+		},
+		{
+			name:           "substution inside escaped percent signs",
+			format:         "%%%s%%",
+			formatArgs:     `"text"`,
+			expectedOutput: "%text%",
+		},
+		{
+			name:           "substitution with one escaped percent sign on the right",
+			format:         "%s%%",
+			formatArgs:     `"percent on the right"`,
+			expectedOutput: "percent on the right%",
+		},
+		{
+			name:           "substitution with one escaped percent sign on the left",
+			format:         "%%%s",
+			formatArgs:     `"percent on the left"`,
+			expectedOutput: "%percent on the left",
+		},
+		{
+			name:           "multiple substitutions",
+			format:         "%d %d %d, %s %s %s, %d %d %d, %s %s %s",
+			formatArgs:     `1, 2, 3, "A", "B", "C", 4, 5, 6, "D", "E", "F"`,
+			expectedOutput: "1 2 3, A B C, 4 5 6, D E F",
+		},
+		{
+			name:           "percent sign escape sequence support",
+			format:         "\u0025\u0025escaped \u0025s\u0025\u0025",
+			formatArgs:     `"percent"`,
+			expectedOutput: "%escaped percent%",
+		},
+		{
+			name:           "fixed point formatting clause",
+			format:         "%.3f",
+			formatArgs:     "1.2345",
+			expectedOutput: "1.234",
+		},
+		{
+			name:           "binary formatting clause",
+			format:         "this is 5 in binary: %b",
+			formatArgs:     "5",
+			expectedOutput: "this is 5 in binary: 101",
+		},
+		{
+			name:           "octal formatting clause",
+			format:         "%o",
+			formatArgs:     "11",
+			expectedOutput: "13",
+		},
+		{
+			name:           "lowercase hexadecimal formatting clause",
+			format:         "%h is 20 in hexadecimal",
+			formatArgs:     "30",
+			expectedOutput: "1e is 20 in hexadecimal",
+		},
+		{
+			name:           "uppercase hexadecimal formatting clause",
+			format:         "%H is 20 in hexadecimal",
+			formatArgs:     "30",
+			expectedOutput: "1E is 20 in hexadecimal",
+		},
+		{
+			name:           "string support with hexadecimal formatting clause",
```

Comment:
I was expecting cases for both `%h` and `%H` for both string and []byte.

Reply:
Fixed - added the additional cases.